joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -408,16 +357,81 @@ JointTrajectoryController::on_configure(const rclcpp_lifecycle::State & previous
     std::bind(&JointTrajectoryController::feedback_setup_callback, this, _1)
   );
 
-  set_op_mode(hardware_interface::OperationMode::INACTIVE);
-
   return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
 }
 
+// Fill ordered_interfaces with references to the matching interfaces
+// in the same order as in joint_names
+template<typename T>
+bool get_ordered_interfaces(
```

Comment:
I like this functionality very much and I think it should be part of the CM. This means that CM should always assign the interfaces in the same order as configured.

you implemented this probably to have `position` and `velocity` states easily available, although they could be implicitly accessed using the `state_interface_` variable, where indexes are the following:
* index_position = 2*index_joint_state
* index_velocity = 2*index_joint_state + 1

Still, for the clearness, I agree with your approach :+1: 

I just propose to move this function into `ControllerInterface` since I can imaging that it will be interesting for many people using the same approach.

Reply:
I wasn't sure that the state_interface_ order was meant to be ordered following anythingin particular so I preferred to go the safest way.

Yeah I'd like to move this to ControllerInterface. @bmagyar thoughts?

Reply:
I agree, `ControllerInterface` should make this handy function available

Reply:
#114